### ISSUE TITLE
pronto having issues with renamed files

### ISSUE THREAD

**Issue by Zechtitus** [mmozuras/pronto#218]
I ran into issues with pronto where it was commenting on commits which were outside of the differences using the --commit flag

Steps - 
1. Basically have a file in your repo /Test/SomeFile.php
2. Add in some generic php code which contains some lint errors
3. Check into gitlab
4. Add in several more additions to that file which pass lint (pronto will see no errors)
5. Rename that file to /Test/SomeFile2.php
- at this point pronto will lint against everything in the file, including all commits back which really should not be in the scope of the diff

bit of background on what we are doing in our gitlab-ci:
pronto run --runner phpcs --formatters gitlab --exit-code --commit=$(git --no-pager log --pretty=format:%H HEAD^1 --reverse | tail -1)

basically we do a diff of the previous commit to compare against. It seems a rename operation will treat the entire file as different and will comment on every linting error (in our case it commented on commits which were several months back prior to us adding in pronto).

**Comment by Zechtitus** [comment]
is any work happening with this issue? got bit again today and devs were spammed with 500+ emails.

funny from one perspective... but still stemming from the renaming issue above.

**Comment by mknapik** [comment]
Does Pronto try to put a comment on every line for the new commit or a comment to every line in commit in which the line has been (really) changed?
Does the issue appear for other formatters? (e.g. default command line).

**Comment by Schwad** [comment]
@Zechtitus does this happen within any particular runners? Do you think this is isolated to gitlab only?

**Comment by Schwad** [comment]
Okay, here is an explicit 'steps to reproduce' on my end. @Zechtitus please confirm if this is the bug you're concerned about.

Tested with pronto-rubocop runner:


`git checkout -b pronto-testaroo`
`Switched to a new branch 'pronto-testaroo'`
`pronto run`
`#=> nothing`
*add double quoted string unnecessarily*
`git add -A`
`git commit -m 'inserts bad string'`
`pronto run`
`test_file.rb:1 W: Prefer single-quoted strings when you don't need string interpolation or special symbols.`
`git checkout master`
`git merge pronto-testaroo`
`pronto run`
`#=> nothing`
`git checkout pronto-testaroo`
`pronto run`
`#=> nothing`
*rename file to test_file2.rb*
`git add -A`
`git commit -m 'renames file'`
`pronto run`
`test_file2.rb:1 W: Prefer single-quoted strings when you don't need string interpolation or special symbols.`

**Comment by Zechtitus** [comment]
That looks about right
